pkg/resourcemonitor/resourcemonitor_test.go

```diff
@@ -316,10 +317,33 @@ func TestNormalizeContainerDevices(t *testing.T) {
 		}
 
 		sort.Slice(res, func(i, j int) bool {
+			if res[i].ResourceName == res[j].ResourceName {
+				var sbi, sbj strings.Builder
+				for _, id := range res[i].DeviceIds {
+					sbi.WriteString(id)
+				}
+
+				for _, id := range res[i].DeviceIds {
+					sbj.WriteString(id)
+				}
+				return sbi.String() < sbj.String()
```

Comment:
I don't mind too much because it's test code, but other alternatives seen in the wild:
1. deep sort all the nested fields ahead of time
2. compare a pretty print of the nested data structure; of course the pretty printer must ensure traversal of the objects in the correct, stable order (e.g. go-spew does this).

Reply:
Nice tricks. In case we'll bump into this issue again I'll try that